components/automate-ui/src/app/pages/data-feed/data-feed.component.ts

```diff
@@ -0,0 +1,136 @@
+import { of as observableOf,  Observable } from 'rxjs';
+
+import { map, filter } from 'rxjs/operators';
+import { Component, OnInit } from '@angular/core';
+import { MatDialog, MatSnackBar } from '@angular/material';
+
+import { SortDirection } from '../../types/types';
+import { DatafeedService } from '../../services/data-feed/data-feed.service';
+import { TelemetryService } from '../../services/telemetry/telemetry.service';
+import {
+  DeleteDatafeedDialogComponent
+} from 'app/page-components/delete-data-feed-dialog/delete-data-feed-dialog.component';
+import { Destination } from './destination';
+
+export interface FieldDirection {
+  name: SortDirection;
+  url: SortDirection;
+}
+
+@Component({
+  selector: 'app-data-feed',
+  templateUrl: './data-feed.component.html',
+  styleUrls: ['./data-feed.component.scss']
+})
+export class DatafeedComponent implements OnInit {
+  destinations$: Observable<Destination[]> = observableOf([]);
+  errorLoading = false;
+  currentPage = 1;
+  pageSize = 10;
+  sortField: string;
+  sortDir: FieldDirection;
+  permissionDenied = false;
+
+  constructor(
+    private service: DatafeedService,
+    public dialog: MatDialog,
+    public snackBar: MatSnackBar,
+    private telemetryService: TelemetryService
+  ) { }
+
+  ngOnInit() {
+    this.destinations$ = this.service.fetchDestinations();
```

Comment:
This will have to be updated to use NGRX, but I would leave that for a later PR.